src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -493,40 +493,37 @@ static double func_call(Object* ho, int narg, int* err) {
 }
 
 static double guigetval(Object* ho) {
-    PyObject* po = ((Py2Nrn*) ho->u.this_pointer)->po_;
-    nanobind::gil_scoped_acquire lock{};
-    PyObject* r = NULL;
-    PyObject* p = PyTuple_GetItem(po, 0);
-    if (PySequence_Check(p) || PyMapping_Check(p)) {
-        r = PyObject_GetItem(p, PyTuple_GetItem(po, 1));
+    nb::tuple po(((Py2Nrn*) ho->u.this_pointer)->po_);
+    nb::gil_scoped_acquire lock{};
+    nb::float_ x{};
+    if (nb::sequence::check_(po[0]) || nb::mapping::check_(po[0])) {
+        x = po[0][po[1]];
     } else {
-        r = PyObject_GetAttr(p, PyTuple_GetItem(po, 1));
+        x = po[0].attr(po[1]);
     }
-    auto pn = nb::steal(PyNumber_Float(r));
-    double x = PyFloat_AsDouble(pn.ptr());
-    return x;
+    return static_cast<double>(x);
 }
 
 static void guisetval(Object* ho, double x) {
-    PyObject* po = ((Py2Nrn*) ho->u.this_pointer)->po_;
-    nanobind::gil_scoped_acquire lock{};
-    auto pn = nb::steal(PyFloat_FromDouble(x));
-    PyObject* p = PyTuple_GetItem(po, 0);
-    if (PySequence_Check(p) || PyMapping_Check(p)) {
-        PyObject_SetItem(p, PyTuple_GetItem(po, 1), pn.ptr());
+    nb::tuple po(((Py2Nrn*) ho->u.this_pointer)->po_);
+    nb::gil_scoped_acquire lock{};
+    if (nb::sequence::check_(po[0]) || nb::mapping::check_(po[0])) {
+        po[0][po[1]] = x;
     } else {
-        PyObject_SetAttr(p, PyTuple_GetItem(po, 1), pn.ptr());
+        po[0].attr(po[1]) = x;
     }
 }
 
 static int guigetstr(Object* ho, char** cpp) {
-    PyObject* po = ((Py2Nrn*) ho->u.this_pointer)->po_;
-    nanobind::gil_scoped_acquire lock{};
-
-    PyObject* r = PyObject_GetAttr(PyTuple_GetItem(po, 0), PyTuple_GetItem(po, 1));
-    auto pn = nb::steal(PyObject_Str(r));
-    Py2NRNString name(pn.ptr());
-    char* cp = name.c_str();
+    nb::tuple po(((Py2Nrn*) ho->u.this_pointer)->po_);
+    nb::gil_scoped_acquire lock{};
+    nb::str name{};
+    if (nb::sequence::check_(po[0]) || nb::mapping::check_(po[0])) {
+        name = po[0][po[1]];
+    } else {
+        name = po[0].attr(po[1]);
+    }
+    const char* cp = name.c_str();
```

Comment:
Why the additional code to deal with sequences and mappings?

Reply:
Removed